### DISCUSSION TITLE
All https except two URLs?

### DISCUSSION THREAD

**Discussion by asmaloney** [nginx-proxy/nginx-proxy#1950] [Q&A]
I'm using this in combination with [docker-letsencrypt-nginx-proxy-companion](https://github.com/JrCs/docker-letsencrypt-nginx-proxy-companion) to use [Let's Encrypt](https://letsencrypt.org/) SSL certs and it's working great.  Easy to set up and get going.

I have a couple of URLs that have to remain _http_ though.  They are hard-coded into some software that can't be changed.

I've been looking at the nginx config docs and the nginx.tmpl file, but I don't see a way to do this.  The `listen 80` block send everything to _https_ so there's no opportunity to deal with the URLs I want.

Am I reading that correctly?  Is there some nginx magic I can use in one of the vhosts files included in the ssl server block to make this work?

**Comment by kevinsimper** [comment]
I think you should use the solution where you provide your own template to docker-gen :+1:

**Comment by asmaloney** [comment]
Thanks Kevin.  I'll take a look.  I was hoping to get away with not having to modify the template somehow...

**Comment by kevinsimper** [comment]
@asmaloney I understand, but it is fairly straigtforward! Remove that line for the config and the mount that into the container instead, and then it works perfect for you! :smile:

**Comment by asmaloney** [comment]
@kevinsimper It's not about how easy it is to make it work - it's about having _Yet Another Thing_ to maintain.

Once I modify that template I have to monitor changes to the nginx-proxy version and apply them to mine.  That doesn't sound so bad, but I already have 1001 other things to keep track of (and I don't do ops full time), so if I can figure out a way to use it as is, I'd prefer to do that.

**Comment by kevinsimper** [comment]
i think your problem is too specific for this project to support your case, especial when the solution is so easy for you to archive.
